lib/chef/dsl/secret.rb

```diff
@@ -30,16 +30,19 @@ module Secret
       #                         perform the secret lookup
       # @option config [Hash] The configuration that the named service expects
       #
+      # @return result [Object] The response object type is determined by the fetcher. See fetcher documentation
+      # to know what to expect for a given service.
+      #
       # @example
       #
       # This example uses the built-in :example secret manager service, which
       # accepts a hash of secrets.
       #
-      #   value = secret(name: "test1", service: :example, config: { "test1" => "value1" } )
+      #   value = secret(name: "test1", service: :example, config: { "test1" => "value1" })
       #   log "My secret is #{value}"
       #
-      #   value = secret(name: "test1", service: :aws_secrets_manager, config: { "region" => "us-west-1" })
-      #   log "My secret is #{value}"
+      #   value = secret(name: "test1", service: :aws_secrets_manager, config: { region: "us-west-1" })
+      #   log "My secret is #{value.secret_string}"
```

Comment:
Would it make sense to have .to_s be the string?

Reply:
In this case, the secret might live in `secret_string` and it might live in `secret_binary`.  It should be either-or, though, so we could probably resolve it. I'll make that update, and we'll see how it is in practice.

Reply:
Updated